intellij/src/saros/intellij/filesystem/IntelliJReferencePointManager.java

```diff
@@ -0,0 +1,203 @@
+package saros.intellij.filesystem;
+
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.module.ModuleUtil;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.roots.ModuleRootManager;
+import com.intellij.openapi.vfs.VirtualFile;
+import java.util.concurrent.ConcurrentHashMap;
+import org.jetbrains.annotations.NotNull;
+import saros.filesystem.IPath;
+import saros.filesystem.IReferencePoint;
+import saros.filesystem.IResource;
+import saros.filesystem.ReferencePointImpl;
+import saros.intellij.project.filesystem.IntelliJPathImpl;
+import saros.repackaged.picocontainer.annotations.Inject;
+
+/**
+ * The IntelliJReferencePointManager maps an {@link IReferencePoint} reference point to {@link
+ * Module} module
+ */
+public class IntelliJReferencePointManager {
```

Comment:
@tobous I am not really sure about the Intellij module stuff. Could look into this class?